src/main/java/minicraft/screen/CraftingDisplay.java

```diff
@@ -100,6 +98,38 @@ public void tick(InputHandler input) {
 			// check the selected recipe
 			Recipe selectedRecipe = recipes[recipeMenu.getSelection()];
 			if (selectedRecipe.getCanCraft()) {
+				Logger.debug("Crafting "+selectedRecipe.getProduct().toString());
+				if (selectedRecipe.getProduct().equals(Items.get("Workbench"))){
+					AchievementsDisplay.setAchievement("minicraft.achievement.benchmarking",true);
+				}
+				if (selectedRecipe.getProduct().equals(Items.get("Plank"))){
+					AchievementsDisplay.setAchievement("minicraft.achievement.planks",true);
+				}
+				if (selectedRecipe.getProduct().equals(Items.get("Wood Door"))){
+					AchievementsDisplay.setAchievement("minicraft.achievement.doors",true);
+				}
+				if (selectedRecipe.getProduct().equals(Items.get("Rock Sword")) ||
+						selectedRecipe.getProduct().equals(Items.get("Rock Pickaxe")) ||
+						selectedRecipe.getProduct().equals(Items.get("Rock Axe")) ||
+						selectedRecipe.getProduct().equals(Items.get("Rock Shovel")) ||
+						selectedRecipe.getProduct().equals(Items.get("Rock Hoe")) ||
+						selectedRecipe.getProduct().equals(Items.get("Rock Bow")) ||
+						selectedRecipe.getProduct().equals(Items.get("Rock Claymore"))) {
+					AchievementsDisplay.setAchievement("minicraft.achievement.upgrade", true);
+				}
+				for (int i = 7; i < 15; i++) {
+					if (selectedRecipe.getProduct().equals(Items.get("blue clothes")) ||
+							selectedRecipe.getProduct().equals(Items.get("green clothes")) ||
+							selectedRecipe.getProduct().equals(Items.get("yellow clothes")) ||
+							selectedRecipe.getProduct().equals(Items.get("black clothes")) ||
+							selectedRecipe.getProduct().equals(Items.get("orange clothes")) ||
+							selectedRecipe.getProduct().equals(Items.get("purple clothes")) ||
+							selectedRecipe.getProduct().equals(Items.get("cyan clothes")) ||
+							selectedRecipe.getProduct().equals(Items.get("reg clothes"))) {
+						AchievementsDisplay.setAchievement("minicraft.achievement.clothes", true);
+					}
+				}
```

Comment:
I don't understand the point of this for loop.

Reply:
Same thing with the old method I had. I where the other big block of or statement is was a for loop. I forgot to remove the other for loop.

Reply:
👍🏻